service/create/load_balancer.go

```diff
@@ -103,90 +62,54 @@ func (s *Service) createLoadBalancer(input LoadBalancerInput) error {
+		return nil, microerror.MaskAny(fmt.Errorf("masters took too long to get running, aborting: %v", err))
 	}
 
 	if err := lb.RegisterInstances(input.InstanceIDs); err != nil {
-		return microerror.MaskAny(fmt.Errorf("could not register instances with LB: %s", errgo.Details(err)))
+		return nil, microerror.MaskAny(fmt.Errorf("could not register instances with LB: %s", errgo.Details(err)))
 	}
 
 	s.logger.Log("debug", fmt.Sprintf("instances registered with ELB"))
 
-	return nil
+	return lb, nil
 }
 
 func (s *Service) deleteLoadBalancer(input LoadBalancerInput) error {
 	// Delete ELB
-	lb, err := awsresources.NewExistingELB(input.Cluster.Spec.Cluster.Cluster.ID, input.Clients.ELB)
+	lbName, err := loadBalancerName(input.Name, input.Cluster)
 	if err != nil {
 		return microerror.MaskAny(err)
 	}
 
-	if err := lb.Delete(); err != nil {
-		return microerror.MaskAny(err)
+	lb := awsresources.ELB{
+		Name:   lbName,
+		Client: input.Clients.ELB,
 	}
-	s.logger.Log("debug", "deleted ELB")
 
-	hzName, err := hostedZoneName(input.Cluster)
-	if err != nil {
+	if err := lb.Delete(); err != nil {
 		return microerror.MaskAny(err)
 	}
-
-	hz, err := awsresources.NewExistingHostedZone(hzName, input.Clients.Route53)
-	if err != nil {
-		underlying := errgo.Cause(err)
-		switch underlying.(type) {
-		case awsresources.DomainNamedResourceNotFoundError:
-			s.logger.Log("debug", "could not find existing hosted zone, continuing")
-		default:
-			return microerror.MaskAny(err)
-		}
-	}
-
-	// Delete DNS record, if the Hosted Zone was found
-	if hz != nil {
-		apiRecordSet := &awsresources.RecordSet{
-			Client:       input.Clients.Route53,
-			Resource:     lb,
-			Domain:       input.Cluster.Spec.Cluster.Kubernetes.API.Domain,
-			HostedZoneID: hz.ID(),
-		}
-
-		if err := apiRecordSet.Delete(); err != nil {
-			return microerror.MaskAny(err)
-		}
-
-		s.logger.Log("debug", "deleted DNS entry for api")
-
-		etcdRecordSet := &awsresources.RecordSet{
-			Client:       input.Clients.Route53,
-			Resource:     lb,
-			Domain:       input.Cluster.Spec.Cluster.Etcd.Domain,
-			HostedZoneID: hz.ID(),
-		}
-
-		if err := etcdRecordSet.Delete(); err != nil {
-			return microerror.MaskAny(err)
-		}
-
-		s.logger.Log("debug", "deleted DNS entry for etcd")
-	}
+	s.logger.Log("debug", fmt.Sprintf("deleted ELB '%s'", lb.Name))
 
 	return nil
 }
 
-// hostedZoneName removes the first 2 subdomains from the API domain
-// e.g.  apiserver.foobar.aws.giantswarm.io -> aws.giantswarm.io
-func hostedZoneName(cluster awstpr.CustomObject) (string, error) {
-	tmp := strings.SplitN(cluster.Spec.Cluster.Kubernetes.API.Domain, ".", 3)
-
-	if len(tmp) == 0 {
-		return "", microerror.MaskAny(malformedDNSNameError)
+// loadBalancerName combines produces a unique name for the load balancer.
+// It takes the domain name, extracts the first subdomain, and combines it with the cluster name
+func loadBalancerName(domainName string, cluster awstpr.CustomObject) (string, error) {
```

Comment:
On the domain for the workers ELB its in the clustertpr but the format is different. It should be ingress.lorenzo.aws.gigantic.io. It needs to use gigantic.io because the ELB is serving customer data not API traffic.

You can see how its set in kubernetesd here.
https://github.com/giantswarm/kubernetesd/blob/master/service/creator/cluster.go#L326

The ingress prefix for the domain isn't in certificatetpr with the others because we aren't generating a cert. But maybe it should be there for completeness. I used a constant in kubernetesd for this.

https://github.com/giantswarm/kubernetesd/blob/master/service/creator/cluster.go#L48